File: SummerLandBackend/Models/Inventory.cs
namespace SummerLandBackend.Models
{
    public class Inventory
    {
        public int Id { get; set; }

        public int ProductVariantId { get; set; }
        public ProductVariant ProductVariant { get; set; } = null!;

        public int LocationId { get; set; }
        public Location Location { get; set; } = null!;

        public int Quantity { get; set; }

        public int LowStockThreshold { get; set; } = 5;
    }
}

File: summerland/lib/models/inventory.dart
class Inventory {
  final int id;
  final int productVariantId;
  final int locationId;
  final String locationName;
  final int quantity;

  Inventory({
    required this.id,
    required this.productVariantId,
    required this.locationId,
    required this.locationName,
    required this.quantity,
  });

  factory Inventory.fromJson(
      Map<String, dynamic> json,
      ) {
    return Inventory(
      id: json['id'],
      productVariantId: json['productVariantId'],
      locationId: json['locationId'],
      locationName: json['locationName'],
      quantity: json['quantity'],
    );
  }
}